api/src/test/java/org/pmiops/workbench/api/CohortBuilderControllerTest.java

```diff
@@ -42,33 +49,97 @@ public void getCriteriaByTypeAndParentId() throws Exception {
         Criteria criteria = listResponse.getItems().get(0);
         assertThat(criteria.getId()).isEqualTo(1);
         assertThat(criteria.getType()).isEqualTo("ICD9");
-        assertThat(criteria.getCode()).isEqualTo("001.1");
-        assertThat(criteria.getName()).isEqualTo("Cholera");
-        assertThat(criteria.getGroup()).isEqualTo(true);
+        assertThat(criteria.getCode()).isEqualTo("001-139.99");
+        assertThat(criteria.getName()).isEqualTo("Infectious and parasitic diseases");
+        assertThat(criteria.getGroup()).isEqualTo(false);
         assertThat(criteria.getSelectable()).isEqualTo(false);
-        assertThat(criteria.getCount()).isEqualTo(10);
-        assertThat(criteria.getDomainId()).isEqualTo("Condition");
+        assertThat(criteria.getCount()).isEqualTo(0);
+        assertThat(criteria.getDomainId()).isNull();
     }
 
     @Test
     public void searchSubjects() throws Exception {
-        final SearchParameter searchParameter = new SearchParameter();
-        searchParameter.setCode("001.1");
-        searchParameter.setDomainId("Condition");
+        ResponseEntity response = controller
+                .searchSubjects(createSearchRequest("ICD9", "001.1", "Condition") );
 
-        final SearchRequest searchRequest = new SearchRequest();
-        final SearchGroupItem item = new SearchGroupItem();
-        final SearchGroup group = new SearchGroup();
-        item.setType("ICD9");
-        item.setSearchParameters(Arrays.asList(searchParameter));
-        group.add(item);
-        searchRequest.setInclude(Arrays.asList(group));
-
-        ResponseEntity response = controller.searchSubjects(searchRequest);
         assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
 
         SubjectListResponse listResponse = (SubjectListResponse) response.getBody();
         String subject = listResponse.get(0);
-        assertThat(subject.equals("1,1,1"));
+        assertThat(subject).isEqualTo("1,1,1");
+    }
+
+    @Test
+    public void searchSubjects_ConditionOccurrenceParent() throws Exception {
+        ResponseEntity response = controller
+                .searchSubjects(createSearchRequest("ICD9", "001", null) );
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        SubjectListResponse listResponse = (SubjectListResponse) response.getBody();
+        String subject = listResponse.get(0);
+        assertThat(subject).isEqualTo("1,1,1");
+    }
+
+    @Test
+    public void searchSubjects_ProcedureOccurrenceLeaf() throws Exception {
+        ResponseEntity response = controller
```

Comment:
You can make all these test cases shorter with a helper function... like:

assertSubjects(createSearchRequest("...."), "2,2,2", ....)